controllers/object_controls.go

```diff
@@ -1293,13 +1316,13 @@ func TransformToolkit(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n
 
 	if len(config.Toolkit.Env) > 0 {
 		for _, env := range config.Toolkit.Env {
-			setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), env.Name, env.Value)
+			setContainerEnv(mainContainer, env.Name, env.Value)
 		}
 	}
 
 	// configure runtime
 	runtime := n.runtime.String()
-	err = transformForRuntime(obj, config, runtime, "nvidia-container-toolkit-ctr")
+	err = transformForRuntime(obj, config, runtime, mainContainerName)
```

Comment:
Something for a follow-up: Does it make sense to rather have `transformForRuntime` accept a pointer to a container instead of a name? We then don't need to iterate over the containers there to find the right container. (Alternatively we could consider an implementation that uses generics).

Reply:
I agree, I can make the proposed change in a follow-up.